File: py/programmers/디스크 컨트롤러.py
import heapq

def solution(jobs):
    temp = []
    answer = 0
    start, end, count = -1, 0, 0
    while count < len(jobs):
        for job in jobs:
            if start < job[0] <= end:
                comeT, useT = job
                heapq.heappush(temp, [useT, comeT])
        if len(temp) > 0:
            useT, comeT = heapq.heappop(temp)
            start = end
            end = end + useT
            answer += (end - comeT)
            count += 1
        else:
            end+=1
            
    return answer//len(jobs)

File: kotlin/programmers/디스크 컨트롤러.kt
import java.util.*

class Solution {
    fun solution(jobs: Array<IntArray>): Int {
        val priorityQueue = PriorityQueue<Pair<Int, Int>>(Comparator { a, b -> a.second - b.second })
        var totalTime = 0
        var currentTime = 0
        var finishedJobs = 0
        var startTime = -1

        while (finishedJobs < jobs.size) {
            for (job in jobs) {
                if (startTime < job[0] && job[0] <= currentTime) {
                    priorityQueue.add(Pair(job[0], job[1]))
                }
            }
            if (priorityQueue.isNotEmpty()) {
                val (start, duration) = priorityQueue.poll()
                startTime = currentTime
                currentTime += duration
                totalTime += (currentTime - start)
                finishedJobs += 1
            } else {
                currentTime += 1
            }
        }
        return totalTime / jobs.size
    }
}
